src/app/compiler.js

```diff
@@ -115,13 +118,12 @@ function Compiler (editor, renderer, queryParams, handleGithubCall, outputField,
 
     if (missingInputs !== undefined && missingInputs.length > 0) {
       compile(missingInputs);
-    } else if (noFatalErrors && !hidingRHP()) {
-      renderer.contracts(data, editor.getValue());
-      formalVerification.compilationFinished(data);
+    } else if (noFatalErrors) {
+      self.event.trigger('compilationSucceed', [data, editor.getValue()]);
```

Comment:
I think we should have another event `compilationFailed`, or have one event `compilationFinished` where we pass success condition. It would also be good if other components could register that event and retrieve the list of errors as part of that callback.